tools/autograd/gen_trace_type.py

```diff
@@ -458,11 +458,20 @@ def emit_trace_body(f: NativeFunction) -> List[str]:
 )
 
 
-def type_wrapper_name(f: NativeFunction) -> str:
+def type_wrapper_name(f: NativeFunction, key: str = "Default") -> str:
     if f.func.name.overload_name:
-        return f"{cpp.name(f.func)}_{f.func.name.overload_name}"
+        name = f"{cpp.name(f.func)}_{f.func.name.overload_name}"
     else:
-        return cpp.name(f.func)
+        name = cpp.name(f.func)
+
+    # The key argument is only used in gen_variable_type where we need fns per autograd dispatch key.
+    # In gen_trace_type and gen_inplace_view_type where only one fn per native_fn must be generated,
+    # the key argument should not be passed.
+    # We do not append key if it is Default so that generated functions from
+    # before per-dispatch-key derivatives were added retain the same names.
+    if key != "Default":
+        name = name + f"_{key}"
```

Comment:
Why do we actually change the name here? Only one function should be generated anyways and so there won't be any conflict.

Reply:
This function is used in `gen_wrapper_registration` in gen_variable_type.cpp which generates the registrations in VariableType.cpp, we need multiple functions in variableType.cpp and I'm differentiating this by appending the keys (e.g. AddTensorBackward vs AddTensorBackward_AutogradNestedTensor)

Reply:
Ho ok! Sounds good!